### DISCUSSION TITLE
Many Completed Pod

### DISCUSSION THREAD

**Discussion by Zruto** [nextflow-io/nextflow#4335] [Q&A]
I am using nextflow, and the working engine is k8s. A large number of Completed pods have not been recycled in the cluster. What is the reason?
![img_v2_bbe103b6-a006-4f62-95ef-f1abccc579bg](https://github.com/nextflow-io/nextflow/assets/46238825/e162ea43-24b1-4b87-b6d1-9d6cd7a880c0)

**Comment by bentsherman** [comment]
That is unexpected, Nextflow should delete successful pods as defined here: https://github.com/nextflow-io/nextflow/blob/1a68049562d892a852ed09cc85f91349f0f51964/modules/nextflow/src/main/groovy/nextflow/k8s/K8sTaskHandler.groovy#L419

**Reply by Zruto** [reply]
There is another question. Is this error caused by insufficient resources on the master node?
```
[Task monitor] ERROR nextflow.processor.TaskProcessor - Execution aborted due to an unexpected error
java.lang.OutOfMemoryError: unable to create native thread: possibly out of memory or process/resource limits reached
```

**Reply by bentsherman** [reply]
Yes, likely the head node where Nextflow is running doesn't have enough memory. A few GB is usually enough
